### ISSUE TITLE
bug: 'le' label getting set to "nil" instead of "+Inf"

### ISSUE THREAD

**Issue by hairyhenderson** [knyar/nginx-lua-prometheus#15]
Not entirely sure why, but I'm getting some metrics lines like:

```
http_response_size_bytes_bucket{method="GET",status="200",le="nil"} 577
```

This causes Prometheus to error on parsing with:

```
text format parsing error in line 9: expected float as value for 'le' label, got "nil"
```

My nginx config looks like:
```nginx
...
lua_package_path '/usr/local/openresty/lualib/?.lua;;';
lua_shared_dict prometheus_metrics 10M;
init_by_lua_file /srv/metrics_init.lua;
log_by_lua_file /srv/metrics_log.lua;
...
```

`/srv/metrics_init.lua`:
```lua
prometheus = require("prometheus").init("prometheus_metrics")

metric_bytes_sent = prometheus:histogram(
  "http_response_size_bytes",
  "HTTP response body sizes",
  {"method", "status"},
  {100, 1000, 10000, 100000, 1000000, 10000000, 100000000}
)
```

`/srv/metrics_log.lua`:
```lua
local method = ngx.var.request_method
local status = ngx.var.status

local bytes_sent = tonumber(ngx.var.body_bytes_sent) or 0

metric_bytes_sent:observe(bytes_sent, {method, status})
```

From the code it looks like that label _should_ be `+Inf`: https://github.com/knyar/nginx-lua-prometheus/blob/master/prometheus.lua#L372... Not really sure what's going on.

**Comment by hairyhenderson** [comment]
Ok, this was _partly_ my fault - what I left out of the bug report was that I _also_ had a label named `protocol` (`ngx.var.protocol`), which was getting set to `Inf` (!?!?!). I realized this was actually getting set to `nil` because I chose the wrong variable - shoulda been `ngx.var.server_protocol` in this case. Anyway, somehow this threw everything off.

This is obviously still a bug though 😉 

As a workaround, I made sure that none of my labels were `nil`, like:

```lua
local method = ngx.var.request_method or ""
local status = ngx.var.status or "0"

metric_bytes_sent:observe(bytes_sent, {method, status})
```
